protocol/retrieval/api.go

```diff
@@ -0,0 +1,36 @@
+package retrieval
+
+import (
+	"context"
+	"io"
+
+	"gx/ipfs/QmR8BauakNcBa3RbE4nbQu76PDiJgoQgz8AJdhJuiU4TAw/go-cid"
+	"gx/ipfs/QmTu65MVbemtUxJEWgsTtzv9Zv9P8rvmqNA4eG9TrTRGYc/go-libp2p-peer"
+
+	"github.com/filecoin-project/go-filecoin/address"
+)
+
+// API here is the API for a retrieval client.
+type API struct {
+	rc       *Client
+	plumbing retrievalClientPorcelainAPI
+}
+
+type retrievalClientPorcelainAPI interface {
+	MinerGetPeerID(ctx context.Context, minerAddr address.Address) (peer.ID, error)
+	//types.Signer
+}
+
+// NewAPI creates a new API for a retrieval client.
+func NewAPI(rc *Client, minerPorc retrievalClientPorcelainAPI) API {
+	return API{rc: rc, plumbing: minerPorc}
+}
+
+// RetrievePiece retrieves bytes referenced by CID pieceCID
+func (a *API) RetrievePiece(ctx context.Context, pieceCID cid.Cid, minerAddr address.Address) (io.ReadCloser, error) {
```

Comment:
it strikes me that if the caller gets the peerid, which is already in porcelain then we could just delete this api abstraction and surface the retrieval client directly. or i guess since we know more stuff is going to go into retrieval we could keep it, but simplify it by eliminating its dep on porcelain

Reply:
let's leave it for cleanup

Reply:
As non-trivial versions of retrieval will be using porcelain functionality (for example to setup payment channels) I think it makes since to keep wrapping abstractions this way, if only to get us ready for the future.